src/stores/weather.ts

```diff
@@ -1,27 +1,35 @@
 import { defineStore } from "pinia";
-import axios from "axios";
+import { Api } from "../services/api";
+import { ApiDto } from "../models/apiDto";
 
 interface WeatherState {
 	weather: Record<string, any> | null;
 	error: string | null;
 	loading: boolean;
+	api: {
+		fetchWeather: (location: string) => Promise<any>;
+	};
 }
 
+const preferences: ApiDto = {
+	apiKey: import.meta.env.VITE_API_KEY,
+	apiEndpoint: "https://api.openweathermap.org/data/2.5/weather",
+};
+
+const api = new Api(preferences);
+
 export const useWeatherStore = defineStore("weather", {
 	state: (): WeatherState => ({
```

Comment:
With vue 3 new api comes in called 'Composition API'. Same thing is possible in pinia. They recommend to use Setup Stores which are like composables with refs as state, computed**s** as getters and functions as actions. Please read [here](https://pinia.vuejs.org/core-concepts/#Setup-Stores)